lib/iron_bank/error.rb

```diff
@@ -24,9 +24,29 @@ def self.from_response(response)
       klass&.new(response)
     end
 
+    # find error class from body (and when in 200 response)
     def self.from_body(response)
```

Comment:
Please be kind; it's been a good while since I've coded Ruby, but perhaps this could be simplified?
```ruby
    def self.from_body(response)
      return unless CODE_MATCHER.match(response.body.to_s)
      body = response.body

      if !body.is_a?(Hash) || body.key?('results')
        return from_text(body.to_s)
      elsif faultcode = body['faultcode'].to_s.sub(/\Afns:/, '')
        return CODE_CLASSES[faultcode]
      end
    end
```
Existing tests pass (though I was only able to install 2.7 on my M2 mac), and I think it needs some commenting, but the logic seems a bit simpler?

Reply:
@rringler 
Thanks! I think there are less lines of code, but less readable. I want to at least have the comments to explain my rationale. 

I also prefer to not worry about what `CODE_CLASSES[""]` would return. WDYT?

Reply:
i like @rringler code -- it's concise and it does not sacrifice readability (imo) some very minor change:
```rb
def self.from_body(response)
  return unless CODE_MATCHER.match(response.body.to_s)
  body = response.body

  if !body.is_a?(Hash) || body.key?('results')
    from_text(body.to_s)
  elsif faultcode = body['faultcode'].to_s.sub(/\Afns:/, '')
    CODE_CLASSES[faultcode]
  end
end
````